### PR TITLE
fix: skaffold debug does not work for helm deployments with multiple …

### PR DESCRIPTION
**Pull request by doriandekoning**
**Fixes:** #4826 
**Related**:https://github.com/GoogleContainerTools/skaffold/pull/4732 
**Merge before/after**: _None_
**Description**
This mr splits the rendered helm resources in the `skaffold filter` command on `---` which separates resources. Currently only the first resource is modified, this causes the debug command not to work.  In this mr the input on `filter` (the rendered helm resources) is split on `---` and to each resource the debugging transformations are applied.

I'd like to add some tests but there are currently no meaning full tests for the filter command (as far as I can see) which I can extend.

### CODE REVIEW

**Comment by doriandekoning** [comment]
@googlebot I signed it!

File: cmd/skaffold/app/cmd/filter.go [hunk 1]
```diff
@@ -64,7 +65,12 @@ func runFilter(ctx context.Context, out io.Writer, debuggingFilters bool, buildA
 		if err != nil {
 			return err
 		}
-		manifestList := manifest.ManifestList([][]byte{bytes})
+
+		var manifestList manifest.ManifestList
+		for _, manifestStr := range strings.Split(string(bytes), "---") {
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/filter.go:70]: This will catch `---` where it occurs within a string.  It needs to be preceded and followed by newlines.

**Review by briandealwis — COMMENTED** [review]
Thanks for catching this — I'll fix this up and add some further tests.

File: cmd/skaffold/app/cmd/filter.go (hunk 1 above)
Comment (doriandekoning) [reply cmd/skaffold/app/cmd/filter.go:70]: Good catch

**Pull request closed without merging**